core/sdl/sdl_gamepad.h

```diff
@@ -4,28 +4,73 @@
 #include "stdclass.h"
 #include "sdl.h"
 
-template<bool Arcade = false, bool Gamepad = false>
+template<bool Arcade = false, bool Gamepad = false, bool DreamcastControllerUsb = false>
 class DefaultInputMapping : public InputMapping
 {
 public:
 	DefaultInputMapping()
 	{
-		name = "Default";
-		set_button(DC_BTN_Y, 0);
-		set_button(DC_BTN_B, 1);
-		set_button(DC_BTN_A, 2);
-		set_button(DC_BTN_X, 3);
-		set_button(DC_BTN_START, 9);
-
-		set_axis(0, DC_AXIS_LEFT, 0, false);
-		set_axis(0, DC_AXIS_RIGHT, 0, true);
-		set_axis(0, DC_AXIS_UP, 1, false);
-		set_axis(0, DC_AXIS_DOWN, 1, true);
-		set_axis(0, DC_AXIS2_LEFT, 2, false);
-		set_axis(0, DC_AXIS2_RIGHT, 2, true);
-		set_axis(0, DC_AXIS2_UP, 3, false);
-		set_axis(0, DC_AXIS2_DOWN, 3, true);
-		dirty = false;
+		if (!DreamcastControllerUsb)
+		{
```

Comment:
This should go to dreamconn.cpp (or a new file) in a new DreamcastUSB InputMapping class, that is loaded when such a controller is detected instead of the default mapping.

Reply:
Superseded by adding mappings for Dreamcast Controller USB during `input_sdl_init()`